datalib/prepare_webdataset.py

```diff
@@ -4,55 +4,52 @@
 import webdataset as wds
 from datasets import load_dataset
 from huggingface_hub import upload_file
-
-from datalib.mappings import dataset_info
-
-def iter_hf_item(dsname, streaming=False):
-    dinfo = dataset_info[dsname]
-    dconfig = {k: dinfo[k] for k in ['path', 'split', 'name']}
-
-    dataset = load_dataset(
-        streaming=streaming,
-        **dconfig
-    )
-
-    dataset = iter(dataset)
-
-    for idx, item in tqdm(enumerate(dataset)):
-        yield item
-
-
-def generate_wds_samples(dsname):
-    for item in iter_hf_item(dsname):
-        transform_func = dataset_info[dsname]['method']
-        sample = transform_func(item)
-
+import os
+import io
+import argparse
+from datalib.mappings import prepare_local_dataset
+
+def generate_wds_samples(dsname, folder_path, channel_name, language):
+    channel_path = os.path.join(folder_path, channel_name)
+    audio_path = os.path.join(channel_path, "audio_files_compressed/")
+    
+    for file in os.listdir(audio_path):
+        filename = os.path.basename(file).replace(".wav", "")
+        sample = prepare_local_dataset(dsname=dsname, channel_name=channel_name, audio_name=filename, folder_path=folder_path, split='train', language=language)
         yield sample
 
-if __name__ == '__main__':
-    import os
-    import argparse
+def process_all_samples(dsname, folder_path, channel_name, language):
```

Comment:
This is again processing all the samples as a list. Read the previous code to see how to use a generator